cpp/oneapi/dal/algo/svm/backend/utils.hpp

```diff
@@ -40,108 +42,151 @@ template <typename Float>
-            new_label_data[i] = in_binary_labels.second;
+    for (std::int64_t i = 1; i < count; ++i) {
+        if (arr_label_host[i] != first_label) {
+            second_label = arr_label_host[i];
             break;
         }
     }
-    if (value_first_class_label == value_second_class_label) {
-        throw invalid_argument{
-            error_msg::input_labels_contain_only_one_unique_value_expect_two()
-        };
+
+    if (first_label == second_label) {
+        throw invalid_argument(
+            dal::detail::error_messages::input_labels_contain_only_one_unique_value_expect_two());
+    }
+
+    // need to sort class labels to be consistent with Scikit-learn*
+    if (first_label > second_label) {
+        std::swap(first_label, second_label);
     }
 
-    for (; i < count; ++i) {
-        if (arr_label_host[i] == value_first_class_label) {
-            new_label_data[i] = in_binary_labels.first;
+    return { first_label, second_label };
+}
+
+template <typename Float>
+inline table get_new_labels(sycl::queue& queue,
+                            const table& labels,
+                            const binary_label_t<Float>& requested_unique_labels,
+                            const binary_label_t<Float>& unique_labels) {
+    auto arr_label = row_accessor<const Float>{ labels }.pull(queue);
+    const std::int64_t count = arr_label.get_count();
+
+    auto new_label_arr = array<Float>::empty(queue, count, sycl::usm::alloc::host);
+    auto new_label_data = new_label_arr.get_mutable_data();
+
+    for (std::int64_t i = 0; i < count; ++i) {
+        if (arr_label[i] == unique_labels.first) {
+            new_label_data[i] = requested_unique_labels.first;
         }
-        else if (arr_label_host[i] == value_second_class_label) {
-            new_label_data[i] = in_binary_labels.second;
+        else if (arr_label[i] == unique_labels.second) {
+            new_label_data[i] = requested_unique_labels.second;
         }
         else {
-            throw invalid_argument{
-                error_msg::input_labels_contain_wrong_unique_values_count_expect_two()
-            };
+            throw invalid_argument(dal::detail::error_messages::
+                                       input_labels_contain_wrong_unique_values_count_expect_two());
         }
     }
 
-    out_binary_labels.first = value_first_class_label;
-    out_binary_labels.second = value_second_class_label;
+    return homogen_table::wrap(queue, new_label_data, count, 1);
+}
 
-    return dal::backend::to_device_sync(queue, new_label_arr);
+template <typename Float>
+inline table convert_binary_labels(sycl::queue& queue,
+                                   const table& labels,
+                                   const binary_label_t<Float>& requested_unique_labels,
+                                   binary_label_t<Float>& unique_labels) {
+    unique_labels = get_unique_labels<Float>(queue, labels);
+    if (unique_labels == binary_label_t<Float>{ 0, 1 } ||
```

Comment:
This if can be moved into `get_new_labels`, which can be renamed to `convert_binary_labels` while this function can be removed. We can require users to call `get_unique_labels` explicitly before the call of that function.